packages/web-components/src/viewing-booking/client/components/viewing-booking.svelte

```diff
@@ -0,0 +1,186 @@
+  import { validateEmail } from '../../../common/utils/validate'
+
+  export let theme
+  export let apiKey
+  export let parentSelector
+  export let submitAction
+
+  let isModalOpen = true
+  let isLoading
+
+  let email
+  let inputValue
+
+  let propertyData
+
+  let backgroundImage
+  let correctEmail = true
+
+  const themeClasses = generateThemeClasses(theme, parentSelector)
+  const { globalStyles } = themeClasses
+
+  function handleToggleModal() {
+    isModalOpen = !isModalOpen
+  }
+
+  function handleInput({ target }) {
+    inputValue = target.value
+    viewBookingStore.update(values => ({
+      ...values,
+      email: inputValue,
+    }))
+  }
+
+  function submitForm() {
+    correctEmail = validateEmail(email)
+    correctEmail && submitAction && submitAction(email)
+  }
+
+  const unsubscribe = viewBookingStore.subscribe(store => {
+    isLoading = store.isLoading
+    email = store.email
+    propertyData = store.propertyData
+  })
+
+  function updateImage(propertyData) {
+    backgroundImage = css`
+      background: url(${propertyData && propertyData.image});
+    `
+  }
+  $: updateImage(propertyData)
+
+  onMount(async () => {
+    isLoading = true
+    const propertyData = await getProperty()
+
+    viewBookingStore.update(values => ({
+      ...values,
+      initializers: {
+        theme,
+        apiKey,
+        parentSelector,
+      },
+      themeClasses,
+      propertyData,
+    }))
+    isLoading = false
+  })
+
+  onDestroy(() => unsubscribe())
+</script>
+
+<style>
+  .viewing_booking-btn {
+    padding: 0.5em;
+    font-size: 1em;
+  }
+  .viewing-booking-email-form {
+    padding: 0.5em 1em;
+  }
+  .viewing-booking-email-form::after {
+    content: '';
+    clear: both;
+    display: block;
+  }
+  .viewing-booking-email-form p {
+    padding: 0.5em 0;
+    line-height: 1em;
+  }
+  .viewing-booking-email-form label {
+    float: left;
+    padding: 0.5em 0;
+    line-height: 1em;
+  }
+  .viewing-booking-email-form input {
+    width: 80%;
+    float: right;
+    height: 2em;
+    border-radius: 3px;
+    border: 1px solid #ddd;
+    outline: none;
+    padding: 0 0.5em;
+    font-size: 1em;
+  }
+  .property-image {
+    background-size: cover;
+    padding: 1em;
+    padding-top: 62.5%;
+    position: relative;
+  }
+  .property-image h4 {
+    position: absolute;
+    bottom: 0;
+    left: 0;
+    width: 100%;
+    color: #ffffff;
+    padding: 0.5em;
+    box-sizing: border-box;
+    font-size: 1.1em;
+  }
+  .property-image h4 strong {
+    float: right;
+    font-weight: 600;
+  }
+  .viewing-booking-form-submit {
+    width: 100%;
+    padding: 0.25em 1em 1em;
+    box-sizing: border-box;
+  }
+  .viewing-booking-form-submit button {
+    width: 100%;
+    padding: 0.3em;
+    border-radius: 3px;
+    background: #f2f2f2;
+    border: 1px solid #ddd;
+    font-size: inherit;
+    outline: none;
+  }
+  .viewing-booking-form-submit button:hover {
+    cursor: pointer;
+    background: #00a7e3;
```

Comment:
There are quite a few colors in this file - can you use the theme for these please? If you look at search widget button there are primary and secondary accent colors for button color / background and inverse for hover if that makes sense?

Reply:
Hi, I applied theme classes to this PR, pls check it again